src/policy/truc_policy.cpp

```diff
@@ -9,32 +9,34 @@
 #include <logging.h>
 #include <tinyformat.h>
 #include <util/check.h>
+#include <util/hasher.h>
 
 #include <algorithm>
 #include <numeric>
 #include <vector>
 
-/** Helper for PackageTRUCChecks: Returns a vector containing the indices of transactions (within
- * package) that are direct parents of ptx. */
-std::vector<size_t> FindInPackageParents(const Package& package, const CTransactionRef& ptx)
+std::vector<std::vector<size_t>> BuildInPackageRelations(const Package& package)
```

Comment:
Would you consider using DepGraph instead of a matrix of indices (it's basically the same thing but more compact)?

Reply:
fwiw I don't think this needs to be direct parents, it's just implemented this way because the limit is 2.

As a more general note - thanks for working on this and I don't want to be discouraging, but I'm not sure that a marginal improvement to this small function is worth this much effort. I see you have some other PRs that might be more impactful, so maybe focus on those? Maybe we can come back to this in the future if it helps with other things.